public/app/core/services/toggleTheme.ts

```diff
@@ -0,0 +1,43 @@
+import { getTheme } from '@grafana/ui';
+import { ThemeChangedEvent } from 'app/types/events';
+import appEvents from '../app_events';
+import { config } from '../config';
+import { PreferencesService } from './PreferencesService';
+
+export async function toggleTheme(runtimeOnly: boolean) {
+  const currentTheme = config.theme;
+  const newTheme = getTheme(currentTheme.isDark ? 'light' : 'dark');
+  appEvents.publish(new ThemeChangedEvent(newTheme));
+
+  if (runtimeOnly) {
+    return;
+  }
+
+  // Add css file for new theme
+  const newCssLink = document.createElement('link');
+  newCssLink.rel = 'stylesheet';
+  newCssLink.href = config.bootData.themePaths[newTheme.type];
+  document.body.appendChild(newCssLink);
```

Comment:
Should we consider using smth like [react-helmet](https://github.com/nfl/react-helmet) for more declarative document `head` updates?

Reply:
Yes, now that the app is fully react we should definitely do that (Especially for updating page title) !!